### PR TITLE
[api] make package release atomic

### PR DESCRIPTION
**Pull request by adrianschroeter**
The real problem behinds coolos publisher penalty issue ....

No test case here, since the behaviour could be only tested in backend. The basic release test is enough as does-not-crash test.

### CODE REVIEW

File: src/api/app/models/project.rb [hunk 1]
```diff
@@ -1417,6 +1417,15 @@ def reopen_release_targets
     Backend::Connection.post "/build/#{URI.escape(name)}?cmd=wipe"
   end
 
+  # lock the project for the scheduler for atomic change when using multiple operations
+  def suspend_scheduler
+    Backend::Connection.post "/build/#{URI.escape(name)}?cmd=suspendproject"
```
Comment (mdeniz) [inline src/api/app/models/project.rb]: Can you make use of the Backend::Api calls? I mean, add those methods there and use the ConnectionHelper methods that we are using there too.

Comment (adrianschroeter) [reply src/api/app/models/project.rb]: that way?

Comment (mdeniz) [reply src/api/app/models/project.rb]: Yes! Thanks 💋

File: src/api/lib/backend/api/build/project.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+module Backend
+  module Api
+    module Build
+      # Class that connect to endpoints related to projects
```
Comment (mdeniz) [inline src/api/lib/backend/api/build/project.rb]: ...related to project builds

👍

**Comment by hennevogel** [comment]
hakiri hangs...

**Pull request merged**